### ISSUE TITLE
Doesn't respect height of view

### ISSUE THREAD

**Issue by millenbop** [mapbox/mapbox-gl-native#4440]
In `3.2.0 beta 2` on iOS, the map view doesn't seem to respect the height of the view, meaning it doesn't show the full world map and the centre is off. So when zoomed in and you set the centre coordinate, it will not be visible.

The following screens illustrate the problem.

![img_2117](https://cloud.githubusercontent.com/assets/12835674/13985652/46464c32-f0fe-11e5-9697-a68fe65ed8be.PNG)
3.0.1

![img_2116](https://cloud.githubusercontent.com/assets/12835674/13985653/46465e34-f0fe-11e5-8bf4-90891ead3471.PNG)
3.2.0 beta 2

The actual `MGLMapView` is of correct size in both instances according to `frame.size.height`.

**Comment by 1ec5** [comment]
What is the map view's `contentInset` at the time that this issue occurs? I wonder if it's incorrectly accounting for the top bar, even though the top bar doesn't overlap the map view.

Also, is the map view initialized with a non-zero-area frame?

**Comment by millenbop** [comment]
```
NSLog(@"%f", self.webView.frame.size.height);
NSLog(@"%f", self.mapView.frame.size.height);
UIEdgeInsets insets = self.mapView.contentInset;
NSLog(@"%f %f %f %f", insets.top, insets.bottom, insets.left, insets.right);

257.000000
257.000000
-297.000000 0.000000 0.000000 0.000000
```

The map is initialized via IB.

**Comment by 1ec5** [comment]
Tagging with 3.2.0 milestone per #4476.

**Comment by friedbunny** [comment]
This was introduced in v3.1.0.

**Comment by friedbunny** [comment]
This bug has to do with the map view’s `.contentInsets` having negative values — only positive values should affect the positioning of features within the map. This affects centering, setting visible coordinates, tracking the user location annotation, and so on.

![simulator-screen-shot-mar-25 -2016 -7 52 41-pm](https://cloud.githubusercontent.com/assets/1198851/14056506/53dbcce8-f2c3-11e5-8aba-1ac22b97d4d1.png)

I think we’re going to have to check every case where we use `.contentInsets` and then discard negative values, where appropriate.

```diff
diff --git a/platform/ios/src/MGLMapView.mm b/platform/ios/src/MGLMapView.mm
index 9b094ec..3224b2b 100644
--- a/platform/ios/src/MGLMapView.mm
+++ b/platform/ios/src/MGLMapView.mm
@@ -1918,7 +1918,7 @@ mbgl::Duration MGLDurationInSeconds(NSTimeInterval duration)
     
     [self willChangeValueForKey:@"visibleCoordinateBounds"];
     mbgl::EdgeInsets padding = MGLEdgeInsetsFromNSEdgeInsets(insets);
-    padding += MGLEdgeInsetsFromNSEdgeInsets(self.contentInset);
+    padding += MGLEdgeInsetsFromNSEdgeInsets([self edgePaddingForCoordinateBounds]);
     mbgl::AnnotationSegment segment;
     segment.reserve(count);
     for (NSUInteger i = 0; i < count; i++)
@@ -1950,6 +1950,14 @@ mbgl::Duration MGLDurationInSeconds(NSTimeInterval duration)
     [self didChangeValueForKey:@"visibleCoordinateBounds"];
 }
 
+/// Returns the edge padding to apply when fitting the map to coordinate bounds.
+- (UIEdgeInsets)edgePaddingForCoordinateBounds
+{
+    UIEdgeInsets insets = self.contentInset;
+    return UIEdgeInsetsMake((insets.top > 0 ? insets.top : 0), (insets.left > 0 ? insets.left : 0),
+                            (insets.bottom > 0 ? insets.bottom : 0), (insets.right > 0 ? insets.right : 0));
+}
+
```

**Comment by friedbunny** [comment]
Fixed by #4504 and its dozen rebases.
